pkg/registry/generic/registry/store.go

```diff
@@ -816,12 +790,6 @@ func (e *Store) filterAndDecorateFunction(m generic.Matcher) func(runtime.Object
 			glog.Errorf("unable to match watch: %v", err)
 			return false
 		}
-		if matches && e.Decorator != nil {
```

Comment:
We need some way like reflect to parse list object.
For example:
```Go
// PodList is a list of Pods.
type PodList struct {
	unversioned.TypeMeta `json:",inline"`
	unversioned.ListMeta `json:"metadata,omitempty"`

	Items []Pod `json:"items"`
}
```
List(pods..) => podList
We need to call Decorate on each of Items and set Items in PodList